feat: update guide for next version (#163)
sapphiredev/website: docs/Guide/getting-started/creating-a-basic-context-menu-command.mdx

---
sidebar_position: 3
title: Creating a basic context menu command
---

> Unlike creating the `Slash Command` without overriding the `Command` class, we need to override the
> `registerApplicationCommands` always to create a `contextMenu` command. For example -

Reviewer: ```suggestion
In order to register Context Menu Commands, we first need to override the `registerApplicationCommands` method. For example:
```

`Example`

> ```typescript ts2esm2cjs|{4-13}|{4-13}
> import { Command } from '@sapphire/framework';
> 
> export class PingCommand extends Command {
>   public constructor(context: Command.Context, options: Command.Options) {
>     super(context, { ...options });
>   }
> 
>   public override registerApplicationCommands(registry: ApplicationCommandRegistry) {
>     registry.registerContextMenuCommand({
>       name: 'ping',
>       type: 'MESSAGE'
>     });
>   }
> }
> ```

Reviewer: ```suggestion
```typescript ts2esm2cjs|{5-14}|{5-14}
import { Command } from '@sapphire/framework';
import { ApplicationCommandType } from 'discord-api-types/v9';

export class UserCommand extends Command {
	public constructor(context: Command.Context, options: Command.Options) {
		super(context, { ...options });
	}

	public override registerApplicationCommands(registry: Command.Registry) {
		registry.registerContextMenuCommand((builder) =>
			builder
				.setName('ping')
				.setType(ApplicationCommandType.Message)
		);
	}
}
```
```

> The above will create a `Context Menu` command called `ping` of `MESSAGE` type.

Reviewer: ```suggestion
The above will create a `Context Menu` command called `ping` of `ApplicationCommandType.Message` type.
```

## Creating the `contextMenuRun` method

To handle the command callback when the context menu command is ran we need to override the `contextMenuRun` method of
command class.

```typescript ts2esm2cjs|{8-17}|{9-17}
import { Command } from '@sapphire/framework';
import type { Message } from 'discord.js';

export class PingCommand extends Command {
  public constructor(context: Command.Context, options: Command.Options) {
    // ...
  }

>   public async contextMenuRun(interaction: Command.ChatInputInteraction) {

Reviewer: ```suggestion
  public async contextMenuRun(interaction: Command.ContextMenuInteraction) {
```

    return interaction.reply('Pong');
  }

>   public override registerApplicationCommands(registry: ApplicationCommandRegistry) {

Reviewer: ```suggestion
  public override registerApplicationCommands(registry: Command.Registry) {
```

---
sidebar_position: 3
title: Creating a basic context menu command
---

In order to register Context Menu Commands, we first need to override the `registerApplicationCommands` method. For
example:

`Example`

```typescript ts2esm2cjs|{5-14}|{5-14}
import { Command } from '@sapphire/framework';
import { ApplicationCommandType } from 'discord-api-types/v9';

export class UserCommand extends Command {
  public constructor(context: Command.Context, options: Command.Options) {
    super(context, { ...options });
  }

  public override registerApplicationCommands(registry: Command.Registry) {
    registry.registerContextMenuCommand((builder) =>
      builder //
        .setName('ping')
        .setType(ApplicationCommandType.Message)
    );
  }
}
```

The above will create a `Context Menu` command called `ping` of `ApplicationCommandType.Message` type.

## Creating the `contextMenuRun` method

To handle the command callback when the context menu command is ran we need to override the `contextMenuRun` method of
command class.

> ```typescript ts2esm2cjs|{8-17}|{9-17}

Reviewer: ```suggestion
```typescript ts2esm2cjs|{8-10}|{9-11}
```
